src/actions/UndoRedo.js

```diff
@@ -0,0 +1,73 @@
+import _isEqual from "lodash/isEqual";
+import { ActionCreators } from "redux-undo";
+
+/*
+ We can't use plain redux-undo actions because some material modifications (eg. supercell creation)
+ require dispatching multiple actions sequentially.
+
+ So we need to manually go through actions history and find the "starting point" of the action we want to undo/redo.
+ "Starting point" is the action that has not the same materials as in present state.
+*/
+
+function getMaterialHash(material) {
+    const hash = material.calculateHash();
+    return hash;
+}
+
+export function undo() {
```

Comment:
This file is not needed, the problem is in how the update is triggered by on changes to basis text in code mirror